src/zope/server/serverchannelbase.py

```diff
@@ -103,9 +105,11 @@ def kill_zombies(self):
         """
         now = time.time()
         cutoff = now - self.adj.channel_timeout
-        for channel in self.active_channels.values():
+        # channel.close calls channel.del_channel, which can change
+        # the size of the map.
+        for channel in list(self.active_channels.values()):
```

Comment:
Ooh, subtle!  Good catch!

Reply:
It's less subtle when it blows up in your face 😄 One of the new tests triggered that.